src/EditorFeatures/Core/Shared/Utilities/WaitContextAdapter.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using Microsoft.CodeAnalysis.Editor.Host;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Shared.Utilities
+{
+    /// <summary>
+    /// An adapter between Roslyn's IWaitContext and editor's IWaitableUIOperationScope, which are basically 
+    /// represent the same abstraction. The only place where it's needed so far is FindReferencesCommandHandler,
+    /// which operates within IWaitableUIOperationScope, but calls to <see cref="IFindReferencesService"/>, which
+    /// requires Roslyn's IWaitContext.
+    /// Going forward this adapter can be deleted once Roslyn's IWaitContext/Indicator is retired in favor of editor's 
```

Comment:
crefs?

Reply:
done

---
In reply to: [158573958](https://github.com/dotnet/roslyn/pull/23769#discussion_r158573958) [](ancestors = 158573958)